TypedSyntax/src/show.jl

```diff
@@ -113,12 +113,40 @@ function type_annotation_mode(node, @nospecialize(T); type_annotations::Bool, hi
     return type_annotate, pre, pre2, post
 end
 
+"""
+    type_string(type)
+
+Overload this function to implement custom annotations for types. For example,
+`type_string(::Type{Float32}) = "F32"` would result in `F32` being used to annotate
+any instance of a `Float32`.
+"""
+function type_string(T)
+    if T isa DataType
+        if isempty(T.parameters)
+            return string(T)
+        else
+            wrapper = Base.typename(T).wrapper
+            return string(wrapper, '{', join(map(type_string, T.parameters), ','), '}')
```

Comment:
I'm a bit wary of this code mapping over `T.parameters` as there were quite a few issues doing a similar thing elsewhere. 
In that code we had to check for whether each element in `T.parameters` was assigned, I don't know if that's a concern here or not.
See https://github.com/JuliaDebug/Cthulhu.jl/pull/508/files for that code and some issues with previous versions of that code are #491, #492, #494.